src/storage/lock_manager.rs

```diff
@@ -9,6 +9,53 @@ pub struct Lock {
     pub hash: u64,
 }
 
+/// Time to wait for lock released when encountering locks.
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum WaitTimeout {
+    None,
+    Default,
+    Millis(u64),
+}
+
+impl WaitTimeout {
+    pub fn expect_millis(self) -> u64 {
+        match self {
+            WaitTimeout::Millis(i) => i,
+            _ => panic!("expected WaitTimeout::Millis"),
+        }
+    }
+
+    pub fn update_if_greater(&mut self, other: u64) {
```

Comment:
I'm still not very satisfied with this name...
What about just providing a single function like `to_millis_or_default(default: u64) -> u64`?

Reply:
This is a good idea! I ended up taking it a step further and returning a `Duration`. And then I noticed the None case and ended up refactoring the type. See the last commit.